templates/common/_base/units/machine-config-daemon-pull.service.yaml

```diff
@@ -0,0 +1,36 @@
+name: "machine-config-daemon-pull.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Machine Config Daemon Pull
+  # Make sure it runs only on OSTree booted system
+  ConditionPathExists=/run/ostree-booted
+  # This "stamp file" is unlinked when we complete
+  # machine-config-daemon-firstboot.service
+  ConditionPathExists=/etc/ignition-machine-config-encapsulated.json
+  Wants=network-online.target
+  After=network-online.target
+  After=rhcos-growpart.service
+
```

Comment:
wondering if we need `After=ignition-firstboot-complete.service` too

Reply:
Good question!  We've discussed in CoreOS moving that service to the initramfs (because that's where Ignition runs), so the ordering would become redundant at some point anyways.

I can't think of any reason we ordered relating to the firstboot complete before either.  Conceptually what we're doing is a "second stage", but it doesn't require the boot partition to be updated.